homework/list.js

```diff
@@ -1,79 +1,95 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+    var TodoListManager = function () { //constructor
+        this._listNoteContent = [];
+        this._wrapper = document.querySelector('#note-list-wrapper');
+    }
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+    TodoListManager.prototype = { //its prototype
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+        start() {
+                 this.fetchList().then(function(data) {
+                     this.updateList(data);
+                     this.drawList();
+                     this.preloadFirstNote();
+                 }.bind(this));
+                 window.addEventListener('click', this);
+        },
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+        onNoteOpen(event) {
+            if (event.target.classList.contains('note-title')) {
+                var id = event.target.dataset.noteId;
+                var content = this._listNoteContent[id];
+                window.dispatchEvent(new CustomEvent('note-open',
+                    { detail: content }));
+            }
+        },
+
+        preloadFirstNote() {
+            if (this._listNoteContent.length !== 0) {
```

Comment:
If a function body is wrapped by a conditional statement only with one branch, it's better to put the `if...else` in the caller, rather than the callee.